### PR TITLE
feat: onesignal integration

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Pulling changes on this PR should allow you to see push notifications from one signal.
- Tested locally and notifications are sent.
- To be able to test it out from scratch, first, clear your cache then go to the applications tab and click "Delete database" like below.
image

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

**Comment by sshanzel** [comment]
@idoshamun I think I am having cache issues with the build. Do you know a quick way to fix this? I will write the solution on a confluence page afterward.

**Comment by idoshamun** [comment]
@sshanzel I'll look at it later if that's ok :)

**Comment by sshanzel** [comment]
> @sshanzel I'll look at it later if that's ok :)

No rush. We can still set up a test environment without this yet. Thank you!

File: packages/extension/.env
Comment (idoshamun) [inline packages/extension/.env:11]: We need to set production `.env` file as well

File: packages/webapp/.env
Comment (idoshamun) [inline packages/webapp/.env:11]: We need to set Vercel as well with the production setting

File: packages/webapp/public/OneSignalSDKWorker.js
Comment (idoshamun) [inline packages/webapp/public/OneSignalSDKWorker.js:1]: We also need the updater right? At least this is what the docs mention

File: packages/shared/src/contexts/NotificationsContext.tsx
Comment (idoshamun) [inline packages/shared/src/contexts/NotificationsContext.tsx:47]: What's the purpose of this snippet?

File: packages/shared/src/contexts/NotificationsContext.tsx
Comment (idoshamun) [inline packages/shared/src/contexts/NotificationsContext.tsx:9]: We need to change the import of this library to dynamic import to reduce the bundle size.
Let me know if you want more information

File: packages/shared/src/contexts/NotificationsContext.tsx
Comment (idoshamun) [inline packages/shared/src/contexts/NotificationsContext.tsx:58]: I'm not sure, but possibly `user` might always be defined even when you are not logged in. Can you please confirm? If so, we might need to change this condition.

Comment (idoshamun) [inline packages/shared/src/contexts/NotificationsContext.tsx:58]: Another point, before merging this PR we have to decide what to do with the extension. Let's discuss the possibilities. We can have a meeting about it or you can suggest

File: packages/shared/src/contexts/NotificationsContext.tsx
Comment (idoshamun) [inline packages/shared/src/contexts/NotificationsContext.tsx:80]: I think "available" might be confusing with whether the user approved it or not

```suggestion
      get isNotificationSupported() {
```

**Review by idoshamun — CHANGES_REQUESTED** [review]
Please see my comments. I think we have more work to do before we can merge it.
Overall I love the structure of how you decided to implement it

File: packages/shared/src/contexts/NotificationsContext.tsx
Comment (sshanzel) [reply packages/shared/src/contexts/NotificationsContext.tsx:80]: I agree. Let's do this.

File: packages/shared/src/contexts/NotificationsContext.tsx
Comment (sshanzel) [reply packages/shared/src/contexts/NotificationsContext.tsx:47]: This used to be an abstraction of the actual `requestPermission`, there are areas where it waits for a response on what actually happened but we can turn this though into a boolean to keep it simple.

File: packages/webapp/public/OneSignalSDKWorker.js
Comment (sshanzel) [reply packages/webapp/public/OneSignalSDKWorker.js:1]: For new implementations, like us, we shouldn't need it anymore. Also tested locally if the notification still works, and thankfully it still does.
image

File: packages/shared/src/contexts/NotificationsContext.tsx
Comment (sshanzel) [reply packages/shared/src/contexts/NotificationsContext.tsx:58]: Surely. I am planning to finish off the webapp first, so I applied some conditions to ensure the extension won't be affected in any way after webapp implementation. Once done, I will go back to the extension and see what can work.

Comment (idoshamun) [reply packages/shared/src/contexts/NotificationsContext.tsx:58]: So I'll keep it as a blocker. We can't merge it before the extension support

File: packages/shared/src/contexts/NotificationsContext.tsx
Comment (sshanzel) [reply packages/shared/src/contexts/NotificationsContext.tsx:47]: This is for unsubscribing. Let me turn this to a boolean value to return so it will be clearer for the consuming end.

File: packages/shared/src/contexts/NotificationsContext.tsx
Comment (sshanzel) [reply packages/shared/src/contexts/NotificationsContext.tsx:58]: This is currently the condition in our `AuthContext` when deciding whether `user` should be null or not.

image

File: packages/webapp/.env
Comment (sshanzel) [reply packages/webapp/.env:11]: Alright. Let me add this now for production setting.

File: packages/shared/src/contexts/NotificationsContext.tsx
Comment (sshanzel) [reply packages/shared/src/contexts/NotificationsContext.tsx:58]: Cool. Would that be a blocker for this PR only or as a whole in the notifications intermediate branch?

File: packages/webapp/.env
Comment (sshanzel) [reply packages/webapp/.env:11]: We might need another OneSignal App for our staging/preview environment as the URL is different from the webapp production URL.

Comment (idoshamun) [reply packages/webapp/.env:11]: I'll create one for our preview deployment and send you so you can set it on Vercel

File: packages/shared/src/contexts/NotificationsContext.tsx
Comment (idoshamun) [reply packages/shared/src/contexts/NotificationsContext.tsx:58]: Looks great!

Comment (idoshamun) [reply packages/shared/src/contexts/NotificationsContext.tsx:58]: For this PR

File: packages/webapp/.env
Comment (sshanzel) [reply packages/webapp/.env:11]: Thank you! 🚢

Comment (idoshamun) [reply packages/webapp/.env:11]: Thinking about it, we can't set a different one for the preview as it works against our production backend.
My suggestion would be to temporarily change the url of production for the testing purposes and then revert. How does it sound?

Comment (sshanzel) [reply packages/webapp/.env:11]: Sounds good to me. This will be a one time testing anyways as there will be not much things to change once setup is successful.

**Review by el — APPROVED** [review]

**Comment by sshanzel** [comment]
@idoshamun feel free to review this PR as well, the missing piece here was the extension-related config which is in a separate PR, but the comments from the initial review should have been fixed here. Thank you for tirelessly reviewing! 🚢

**Review by idoshamun — APPROVED** [review]
🤩

**Pull request merged**
